src/app-interoperability/application-package-definition.linkml.yaml

```
@@ -0,0 +1,497 @@
+        rank: 10
+      components:
+        range: HelmComponent
+        rank: 20
+
+  DockerComposeDeploymentProfile:
+    is_a: DeploymentProfile
+    rank: 40
+    slot_usage:
+      type:
+        equals_string: "docker-compose"
+        rank: 10
+      components:
+        range: DockerComposeComponent
+        rank: 20
+
+  Component:
+    description: A class representing a component of a deployment profile.
+    rank: 50
+    attributes:
+      name:
+        description: >-
+          A unique name used to identify the component package. For helm installations the name will be used as the chart name.
+          The name must be lower case letters and numbers and MAY contain dashes. 
+          Uppercase letters, underscores and periods MUST NOT be used.
+        required: true
+        range: string
+        rank: 10
+      properties:
+        description: >-
+           A dictionary element specifying the component packages's deployment details. 
+           See the [Component Properties](#component-properties) section below.
+        range: ComponentProperties #Arne: error in spec, as it is defined as map[string][interface{}] 
+        required: true
+        multivalued: true
+        inlined: true
+        inlined_as_list: false
+        rank: 20
+
+  HelmComponent:
+    is_a: Component
+    rank: 50
+
+  DockerComposeComponent:
+    is_a: Component
+    rank: 50
+  
+  ComponentProperties:
+    description: Properties dictionary for component deployment details.
+    rank: 80
+    attributes:
+      repository:
+        description: Repository location for the component.
+        rank: 10
+        range: string
+      revision:
+        description: Revision version for the component.
+        rank: 20
+        range: string
+      wait:
+        description: If True, indicates the device waits for the component installation to complete.
+        rank: 30
+        range: boolean
+      timeout:
+        description: Time to wait for component installation to complete, formatted as "##m##s".
+        rank: 40
+        range: string
+      packageLocation:
+        description: URL indicating the Docker Compose package's location.
+        rank: 50
+        range: string
+      keyLocation:
+        description: URL for the public key used to validate a digitally signed package.
+        rank: 60
+        range: string
+
+  Parameter:
+    description: Defines a configurable parameter for the application.
+    rank: 90
+    attributes:
+      name: #Arne: error in spec. This attribute is missing.
+        description: Name of the parameter.
+        rank: 10
+        identifier: true
+        required: true
+        range: string
+      value:
+        description: >-
+          The parameter's default value. 
+          Accepted data types are string, integer, double, boolean, array[string], array[integer], array[double], array[boolean].
+        rank: 20
+        range: string #Arne: error in spec. type not properly given.
```

@arne-broering value might be string, integer, double, boolean, array[string], array[integer], array[double], array[boolean]; but here only string is being accepted. Let's add support at least for scalar types, support for arrays can remain for the time being TBD
```suggestion
        any_of: #support for arrays still TBD
          - range: boolean
          - range: integer
          - range: double
          - range: string
```